ignite/base/events_driven.py

```diff
@@ -304,14 +308,30 @@ def __setattr__(self, attr: str, value: Any) -> None:
 
         super().__setattr__(attr, value)
 
-    def update_mapping(self, event_to_attr: Mapping[Any, str]) -> None:
-        """Looks for the attribute and check if not existed to add it.
+    def update_mapping(self, attr_to_events: Mapping[str, List["Events"]]) -> None:
+        """Maps each attribute to a list of the corresponding events in the same given order.
 
         Args:
-            event_to_attr: mapping consists of the events from :class:`~ignite.engine.events.Events`
-                or any other custom events added by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
+            attr_to_events: mapping consists of the attributes mapped to a list events from
+                :class:`~ignite.engine.events.Events` or any other custom events added
+                by :meth:`~ignite.base.events_driven.EventsDriven.register_events`.
+                New attributes will be added, existed ones will be updated.
+                Getting attribute values is done based on the first element in the list of the events.
         """
-        for k, v in event_to_attr.items():
-            attr_evnts = self._attr_to_events[v]
-            if k not in attr_evnts:
-                attr_evnts.append(k)
+        if (
+            not isinstance(attr_to_events, dict)
+            or not all(isinstance(key, str) for key in list(attr_to_events.keys()))
+            or not all(isinstance(value, list) for value in list(attr_to_events.values()))
+        ):
+            raise TypeError(
+                "'attr_to_events' must be a dictionary, it shoule map keys of type string to list of Events"
+            )
+
+        for k, v in attr_to_events.items():
+            if k not in self._attr_to_events:
+                self._attr_to_events[k] = v  # type: ignore
+            elif k in self._attr_to_events:
+                attr_evnts = self._attr_to_events[k]
+                for evnt in v:
+                    if evnt not in attr_evnts:
+                        attr_evnts.append(evnt)
```

Comment:
I think we should not alter existing attributes in the map, otherwise the API is cumbersome.